samples/bluetooth/channel_sounding/README.rst

```diff
@@ -0,0 +1,137 @@
+.. zephyr:code-sample:: ble_cs
+   :name: Channel Sounding
+   :relevant-api: bt_gap bluetooth
+
+   Use Channel Sounding functionality.
+
+
+Bluetooth: Channel Sounding
+###########################
+
+Overview
+********
+
+These samples demonstrates how to use the Bluetooth Channel Sounding feature.
+
+The CS Test sample shows how to us the CS test command to override randomization of certain channel
+sounding parameters, experiment with different configurations, or evaluate the RF medium. It can
+be found under :zephyr_file:`samples/bluetooth/channel_sounding/cs_test`.
+
+The connected CS sample shows how to set up regular channel sounding procedures on a connection
+between two devices.
+It can be found under :zephyr_file:`samples/bluetooth/channel_sounding/connected_cs`
+
+A basic distance estimation algorithm is included in both.
+The Channel Sounding feature does not mandate a specific algorithm for computing distance estimates,
+but the mathematical representation described in [#phase_and_amplitude]_ and [#rtt_packets]_ is used
+as a starting point for these samples.
+
+Distance estimation using channel sounding requires data from two devices, and for that reason
+the channel sounding results in the sample are exchanged in a simple way using a GATT characteristic.
+This limits the amount of data that can be processed at once to about 512 bytes from each device,
+which is enough to estimate distance using a handful of RTT timings and PBR phase samples across
+about 35-40 channels, assuming a single antenna path.
+
+Both samples will perform channel sounding procedures repeatedly and print regular distance estimates to
+the console. They are designed assuming a single subevent per procedure.
+
+Diagrams illustrating the steps involved in setting up channel sounding procedures between two
+connected devices are available in [#cs_setup_phase]_ and [#cs_start]_.
+
+Requirements
+************
+
+* Two boards with Bluetooth LE and Channel Sounding support (such as an nRF54L15 devkit from Nordic Semiconductor)
+* A controller that supports the Channel Sounding feature (such as the Nordic SoftDevice Controller)
```

Comment:
This has me wondering a little bit. Ideally we shouldn't need to have references to specific boards, rather there should be an easy way to search for upstream supported boards which support a certain Bluetooth core spec version or specific Bluetooth feature. Should you perhaps at least reference the exact board board target, or have a link to the board's upstream documentation page?

As for the controller reference, I think that's even worse, since you're now referring to something which doesn't exist upstream, or rather directing the user to a downstream proprietary tree. I'm not sure we should be doing such things in the upstream tree. A slightly better situation would be some other proprietary controller which *is* buildable upstream (using `west blobs`), but I'm not sure generic samples should be encouraging that either.

@carlescufi any thoughts here?

Reply:
> Ideally we shouldn't need to have references to specific boards, rather there should be an easy way to search for upstream supported boards which support a certain Bluetooth core spec version or specific Bluetooth feature.

I agree, that would be nice but this is outside of the scope of this PR.

> Should you perhaps at least reference the exact board board target, or have a link to the board's upstream documentation page?

I agree, here is a suggestion @olivier-le-sage:

```suggestion
* Two boards with Bluetooth LE and Channel Sounding support (such as an `nRF54L15 DK`_)
* A controller that supports the Channel Sounding feature
```